### PR TITLE
feat(talos): v2 engine support  

### PR DESCRIPTION
**Pull request by c3y1huang**
#### Which issue(s) this PR fixes:

Issue longhorn/longhorn#7791

#### What this PR does / why we need it:

Remove the dependency on the host's bash shell to support the v2 engine on Talos Linux.

#### Special notes for your reviewer:

`None`

#### Additional documentation or context

`None`

### CODE REVIEW

File: pkg/nvme/initiator.go [hunk 1]
```diff
@@ -80,12 +79,25 @@ func NewInitiator(name, subsystemNQN, hostProc string) (*Initiator, error) {
 	}, nil
 }
 
+func (i *Initiator) getNewLock() (*commonns.FileLock, error) {
```
Comment (derekbit) [inline pkg/nvme/initiator.go:82]: How about `newLock()`?

**Review by derekbit — APPROVED** [review]
LGTM

**Comment by innobead** [comment]
@derekbit Do we have a ticket for SLE Micro support? Or it should be not different from other general OS support except for package installation?

**Comment by derekbit** [comment]
> @derekbit Do we have a ticket for SLE Micro support? Or it should be not different from other general OS support except for package installation?

We don't have ticket for it. I think there is no difference.

**Review by derekbit — APPROVED** [review]

**Pull request merged**